lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -130,6 +134,22 @@ def add_debit(xml, money, options)
         end
       end
 
+      def add_preauth(xml, money, options)
+        currency    = options[:currency] || currency(money)
+        description = options[:description].blank? ? 'Preauthorize' : options[:description]
```

Comment:
I don't think that this is necessary. The transactions seem to work fine without sending anything for the `description` parameter. Instead, consider replacing line 148 below with
`xml. description options[:description]`

Reply:
Do we want to get rid of it in the `purchase` method too? (or has that maybe already happened and my branch was not up to date?)

Reply:
I believe I placed the hardcoded fallback in place because the Ixopay docs indicated that the description is required under certain conditions. I did not find details on what those conditions are, and had also seen this pattern with Worldpay. That said, I'll leave it up to you @therufs.

Reply:
Thanks for clarifying @jasonxp. The Ixopay docs do indeed indicate that the `description` field is "conditionally mandatory". (https://gateway.ixopay.com/documentation/direct-pci-enabled-api?xml#transaction-data)

If that's something that we want to enforce to be on the safe side, then it should probably be wrapped in a test for both the `authorize` and `purchase` methods.

Reply:
I would assume that in the cases when the field is mandatory, a specific value will be required, and we want it to be the user's responsibility to populate the field accordingly, rather than ours -- kind of a built-in GSF. I think it's unlikely that leaving the field in will cause any problems, so I'm not averse to leaving it; and I'm going to go ahead and updated the unit tests to check for it, but am not sure those updates actually give us information about whether we can expect transactions sent to the gateway to succeed.

Reply:
Hmm, that's an interesting point. We don't actually have any idea of what Ixopay would require that field to be, much less when it's required. Maybe this is a good thing to leave up to our customers instead of instilling a false sense of security. 🤷‍♀

Reply:
I actually would expect a description field to be free-form, rather than provide a specific value. We can follow up with Ixopay to find out specifics around the "conditionally mandatory" statement.

Reply:
Right -- I was thinking in cases where it's something like "provide the authorization from the initial transaction" or something for recurring charges.